### PR TITLE
refactor and restructure with removing jose

### PR DESCRIPTION
**Pull request by vijayshiyani**
refactoring and removing `jose` from dependencies.

### CODE REVIEW

File: src/holder.ts [hunk 1]
```diff
@@ -1,61 +1,64 @@
...
    * @param forVerifier The verifier to get the key binding JWT for.
-   * @returns The key binding JWT and nonce.
-   * @throws An error if the key binding JWT could not be created.
+   * @param nonce The nonce to use.
+   * @throws An error if the key binding JWT cannot be created.
+   * @returns The key binding JWT.
    */
-  async getKeyBindingJWT(forVerifier: string): Promise<{ keyBindingJWT: JWT; nonce: string }> {
+  async getKeyBindingJWT(
+    forVerifier: string,
```
Comment (linasi) [inline src/holder.ts:28]: @vijayshiyani what is this `forVerifier`? Is a strng that will be put as `aud` claim?

File: src/types.ts [hunk 2]
```diff
@@ -44,3 +45,123 @@ export interface IssuerMetadata {
...
+  algo: supportedAlgorithm;
+  callback: Signer;
+};
+
+/**
+ * JOSE Types
+ */
+
+/** Recognized JWT Claims Set members, any other members may also be present. */
+export interface JWTPayload {
```
Comment (linasi) [inline src/types.ts:64]: @vijayshiyani I believe some of these types will be added to `sd-jwt` library and some of these definitions will be removed?

**Review by linasi — APPROVED** [review]
👍

File: src/types.ts (hunk 2 above)
Comment (ragnika) [reply src/types.ts:64]: @vijayshiyani I've added the following types
`JWK`
`JWTHeaderParameters`
`JWTPayload`

you can import these types from the latest `sd-jwt` lib

File: src/holder.spec.ts [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
-  let algorithm: supportedAlgorithm;
 
   beforeEach(async () => {
-    algorithm = supportedAlgorithm.ES256;
-    const keyPair = await generateKeyPair(algorithm);
-    privateKey = keyPair.privateKey;
-    holder = new Holder(privateKey, algorithm);
+    const keyPair = await generateKeyPair(supportedAlgorithm.ES256);
+    holder = new Holder({
+      algo: supportedAlgorithm.ES256,
```
[the remark this answers is not in the document]
Comment (vijayshiyani) [reply src/holder.spec.ts]: renamed to `alg`

File: src/types.ts (hunk 2 above)
Comment (vijayshiyani) [reply src/types.ts]: I have updated above types to be  used from `sd-jwt`

File: src/holder.spec.ts [hunk 4]
```diff
@@ -1,17 +1,17 @@
...
+      callback: signerCallbackFn(keyPair.privateKey),
```
[the remark this answers is not in the document]
Comment (vijayshiyani) [reply src/holder.spec.ts:13]: @linasi  yes, it's for signing a key-binding token. I will add better documentation for that to make it more clear.

File: src/holder.ts (hunk 1 above)
Comment (vijayshiyani) [reply src/holder.ts:28]: @linasi  yes, renamed to `aud` with comments, it makes more sense.

**Pull request merged**